### DISCUSSION TITLE
Domain name in multi-tier setups

### DISCUSSION THREAD

**Discussion by afflerbach** [coupergateway/couper#272] [Q&A] [answered]
Is it somehow possible to access Couper's actual domain name even when operating in a multi-tier setup?

So far I have:

```hcl
set_response_headers = {
  Set-Cookie = "foo=bar; domain=${request.host}"
}
```

This is fine unless I move Couper behind my SSL offloader as I then of course want the cookie to be set for the offloader's domain.

Any suggestions?

**Comment by johakoch** [comment, marked as answer]
Have a look at couper's [run options/environment variables](https://github.com/avenga/couper/blob/master/docs/CLI.md#run-options). The variable `COUPER_ACCEPT_FORWARDED_URL` configures, which `X-Forwarded-*` request headers modify the [request variables](https://github.com/avenga/couper/blob/master/docs/REFERENCE.md#request) `url`, `origin`, `protocol`, `host`, and `port`.

So in order to get the actual host name in `request.host`, set `COUPER_ACCEPT_FORWARDED_URL=host` and make sure that your SSL offloader sends the `X-Forwarded-Host` request header.

E.g. if the SSL offloader sends `X-Forwarded-Host: www.example.com`, `request.host` will be "www.example.com".

If you also want "https" in `request.url`, `request.origin` and `request.protocol`, set `COUPER_ACCEPT_FORWARDED_URL=proto,host` and let the SSL offloader send `X-Forwarded-Proto` as well.

In a local development setup (without an SSL offloader), you don't set `COUPER_ACCEPT_FORWARDED_URL`, and `request.host` is couper's host name.
